### ISSUE TITLE
Unsupported CTRL-Z makes pwsh session unusable on iTerm 2

### ISSUE THREAD

**Issue by bbhoss** [PowerShell/PowerShell#7854]
Steps to reproduce
------------------

```powershell
tail -f /var/log/system.log
# Hit CTRL-Z
# Hit CTRL-C
```

Expected behavior
-----------------
I would expect it to background the job, as specified in #3229. Since `bg`-like functionality is not supported, I would expect it to do nothing, then allow me to terminate/interrupt the running job with CTRL-C.

Actual behavior
---------------
It seems that after CTRL-Z is hit, the terminal becomes completely unresponsive. This is only on iTerm, it works fine with Kitty and Terminal. I'm hoping someone can help me investigate.

Environment data
----------------

```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      6.1.0
PSEdition                      Core
GitCommitId                    6.1.0
OS                             Darwin 17.7.0 Darwin Kernel Version 17.7.0: Thu Jun 21 22:53:14 PDT 2018; root:xnu-4570.71.2~1/RELEASE_X86_64
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by bbhoss** [comment]
Filed an [issue](https://gitlab.com/gnachman/iterm2/issues/7080) with iTerm as well

**Comment by mklement0** [comment]
I don't see the issue in iTerm2 3.2.1beta5: `^Z` prints when I press `Ctrl-Z`, with no other effect, and `^C` prints when I press `Ctrl-C`, and it terminates the command, as expected.

Just to rule out other factors: Do you have a PSReadLine handler set up for `Ctrl-Z` / `Ctrl-C`?

**Comment by bbhoss** [comment]
It seems CTRL-C is bound to
```
Ctrl+c           CopyOrCancelLine      Either copy selected text to the clipboard, or if no text is selected, cancel editing the line with CancelLine.
```

I tried the latest nightly build of iTerm to no avail.

**Comment by mklement0** [comment]
On second thought: PSReadLine may have nothing to do with this - sorry (PSReadLine is not intercepting keyboard input while a program is running).

I wonder if the problem is related to your specific iTerm2 settings - may be worth trying with the default settings: Rename `"$env:HOME/Library/Application Support/iTerm2"` (temporarily), then start iTerm2 and see if the problem still occurs.

Note: On restoring the original ``"$env:HOME/Library/Application Support/iTerm2"` folder you may have to mark your previous default profile as the default one again.

**Comment by bbhoss** [comment]
This didn't change anything. It also seems to have kept my font around though, so not sure that did what you were expecting. I tried removing the preferences as well to no avail.

**Comment by mklement0** [comment]
> I tried removing the preferences as well to no avail.

Makes sense - my assumption was that renaming the application-support folder would be enough - but I don't know for sure; and I'm not even sure if there are other configuration files somewhere.

So it's only Ctrl-Z that triggers the symptom, Ctrl-C by itself works as expected?

The profile you're using doesn't have Ctrl-Z remapped?

I'm personally out of ideas, except to suggest turning on iTerm2's debug logging via `iTerm2 > Toggle Debug Logging` and studying the logs.
